src/nanoemoji/glue_together.py

```diff
@@ -1,37 +1,82 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Copies color assets from one font to another.
+
+Both must use the same glyph names."""
+from absl import app
+from absl import flags
+from absl import logging
 from fontTools.ttLib.tables import otTables as ot
 from fontTools import ttLib
 import os
 
-_ORIGINAL="/tmp/Bungee7.ttf"
-_COLR="/tmp/bungee7/AnEmojiFamily.ttf"
 
-font = ttLib.TTFont(_ORIGINAL)
-colr = ttLib.TTFont(_COLR)
-_OUTPUT = '/tmp/bungee7/Bungee_SVG_COLR.ttf'
+FLAGS = flags.FLAGS
+
+
+flags.DEFINE_string("target_font", None, "Font assets are copied into.")
+flags.DEFINE_string("donor_font", None, "Font from which assets are copied.")
+flags.DEFINE_string("color_table", None, "The color table to copy.")
+flags.DEFINE_string("output_file", None, "Font assets are copied into.")
+
+
+def _copy_colr(target: ttLib.TTFont, donor: ttLib.TTFont):
+    # Copy all glyphs used by COLR over
+    _glyphs_to_copy = set()
+    new_glyphorder = list(target.getGlyphOrder())
+
+    def _collect_glyphs(paint):
+        if paint.Format == ot.PaintFormat.PaintGlyph:
+            _glyphs_to_copy.add(paint.Glyph)
+
+    for record in donor["COLR"].table.BaseGlyphList.BaseGlyphPaintRecord:
+        record.Paint.traverse(donor["COLR"].table, _collect_glyphs)
+
+    for glyph_name in _glyphs_to_copy:
+        print("Copy glyph", glyph_name)
+        target["glyf"].glyphs[glyph_name] = donor["glyf"].glyphs[glyph_name]
+        target["hmtx"].metrics[glyph_name] = donor["hmtx"].metrics[glyph_name]
 
-# Copy all glyphs used by COLR over
-_glyphs_to_copy = set()
+        # .notdef in particular likely to be in both
+        if glyph_name not in new_glyphorder:
```

Comment:
here is where we should fail (or handle nicely) `if glyph_name in new_glyphorder`

Reply:
Since we've now got an explicit destructive mode I think this is OK? - reopen if you disagree